### ISSUE TITLE
"mvn package" stops with error on Mac OS X when DMG file already exists: "hdiutil: create failed - file exists".

### ISSUE THREAD

**Issue by treimers** [fvarrui/JavaPackager#73]
I'm submitting a feature request.

- [ ] bug report
- [x] feature request
- [ ] other

**Short description of the issue/suggestion:**

When running "mvn package" twice on Mac OS X generation of DMG stops with an error: "hdiutil: create failed - file exists".

**Steps to reproduce the issue/enhancement:**

Simply run "mvn package" twice under Mac OS X.

**What is the expected behavior?**

The existing DMG file should be overwritten.

**What is the current behavior?**

The generation of DMG stops with an error: "hdiutil: create failed - file exists".

**Do you have outputs, screenshots, demos or samples which demonstrate the problem or enhancement?**

```
[INFO]     Packager initialized!
[INFO]     
[INFO]     Generating DMG image...
[INFO]         Creating Applications link
[INFO]         Creating symbolic link [/Users/treimers/Documents/datasqill/git/datasqillj15/workbench/target/deploy/osx/datasqill/Applications] to [/Applications]
[INFO]         Copying background image
[INFO]         Copying file [/Users/treimers/Documents/datasqill/git/datasqillj15/workbench/src/main/deploy/package/mac/datasqill-background.png] to folder [/Users/treimers/Documents/datasqill/git/datasqillj15/workbench/target/deploy/osx/datasqill/.background/background.png]
[INFO]         Copying icon file: /Users/treimers/Documents/datasqill/git/datasqillj15/workbench/src/main/deploy/package/mac/datasqill.icns
[INFO]         Copying file [/Users/treimers/Documents/datasqill/git/datasqillj15/workbench/src/main/deploy/package/mac/datasqill-volume.icns] to folder [/Users/treimers/Documents/datasqill/git/datasqillj15/workbench/target/deploy/osx/datasqill/.VolumeIcon.icns]
[INFO]         Creating image: /Users/treimers/Documents/datasqill/git/datasqillj15/workbench/target/deploy/osx/assets/datasqill_3.4.6.0.dmg
[INFO]         Executing command: /bin/sh -c cd '/Users/treimers/Documents/datasqill/git/datasqillj15/workbench/.' && 'hdiutil' create -srcfolder /Users/treimers/Documents/datasqill/git/datasqillj15/workbench/target/deploy/osx/datasqill -volname datasqill -fs HFS+ -format UDRW /Users/treimers/Documents/datasqill/git/datasqillj15/workbench/target/deploy/osx/assets/datasqill_3.4.6.0.dmg
[ERROR]         hdiutil: create failed - Datei existiert
[ERROR]     DMG image generation failed due to: Command execution failed: hdiutil create -srcfolder /Users/treimers/Documents/datasqill/git/datasqillj15/workbench/target/deploy/osx/datasqill -volname datasqill -fs HFS+ -format UDRW /Users/treimers/Documents/datasqill/git/datasqillj15/workbench/target/deploy/osx/assets/datasqill_3.4.6.0.dmg
```

**What is the motivation / use case for changing the behavior?**

After changes to any code of my application I would like to simply run "mvn package" and not "mvn clean package" because this would take much longer.

**Please tell us about your environment:**

- JavaPackager version: 15
- OS version: Mac OS, El Capitan, 10.11.6
- JDK version: OpenJDK Runtime Environment AdoptOpenJDK (build 15+36)
- Build tool: 
	- [x] Maven - Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
	- [ ] Gradle 

**Other information** (e.g. related issues, suggestions how to fix, links for us to have context)

I did execute the hdiutil command manually and found a possible solution:

```
hdiutil create -srcfolder /Users/treimers/Documents/datasqill/git/datasqillj15/workbench/target/deploy/osx/datasqill -volname datasqill -fs HFS+ -format UDRW /Users/treimers/Documents/datasqill/git/datasqillj15/workbench/target/deploy/osx/assets/datasqill_3.4.6.0.dmg
hdiutil: create failed - file exists
```

There is an option "-ov" to "overwrite" existing files. That helped in my environment

```
hdiutil create -srcfolder /Users/treimers/Documents/datasqill/git/datasqillj15/workbench/target/deploy/osx/datasqill -volname datasqill -ov -fs HFS+ -format UDRW /Users/treimers/Documents/datasqill/git/datasqillj15/workbench/target/deploy/osx/assets/datasqill_3.4.6.0.dmg
```

It would be nice to enhance the DMG creation by adding this option to the command.

Thanks in advance
Thorsten

**Comment by fvarrui** [comment]
Great! Thanks for helping to improve this plugin

**Comment by fvarrui** [comment]
Change applied! It will be published in v1.4.0 asap.

Thanks!

**Comment by treimers** [comment]
Cool, thank you!

**Comment by fvarrui** [comment]
v1.4.0 released to Maven Central

**Comment by fvarrui** [comment]
Hi @treimers!
I realized that there's the same problem later, when compressing the DMG disk image file. It's fixed in v1.4.1-SNAPSHOT.

**Comment by treimers** [comment]
Hi @fvarrui,

yes you are righ I just cheched. And sorry that I did not recognize it :(

Great that you already fixed it, Thanks!
